api/src/main/java/org/pmiops/workbench/api/WorkspacesController.java

```diff
@@ -553,7 +563,20 @@ private void createConfigFile(org.pmiops.workbench.firecloud.model.Workspace fcW
     fireCloudService.cloneWorkspace(workspaceNamespace, workspaceId,
         fcWorkspaceId.getWorkspaceNamespace(), fcWorkspaceId.getWorkspaceName());
 
-    // TODO(calbach): Determine whether we need to copy GCS notebooks here.
+    String toBucket = null;
+    try {
+      toBucket = fireCloudService.getWorkspace(
+          fcWorkspaceId.getWorkspaceNamespace(), fcWorkspaceId.getWorkspaceName())
+          .getWorkspace()
+          .getBucketName();
+    } catch (ApiException e) {
+      log.log(Level.SEVERE, "Firecloud error retrieving newly cloned workspace", e);
+      throw new ServerErrorException();
+    }
+
+    for (String name : cloudStorageService.getBucketFileList(fromBucket, "notebooks")) {
+      cloudStorageService.copyBlob(BlobId.of(fromBucket, name), BlobId.of(toBucket, name));
```

Comment:
What should we do if an individual file copy fails... bail out? Log it / return an error in the response, but copy the rest? (I am not certain.)

For now, bailing out seems OK.

Reply:
Good question. I think we bail out. I added a comment here which describes the state of things if this fails.

Ultimately, I think we do not want to automatically continue if notebook cloning fails. However in the future it might make sense to have an optional CloneRequest parameter which allows you to skip notebook cloning, if that becomes the issue (and some indication in the error response if that was the cause for failure). UX would look like "seems like there's an issue cloning notebooks from this workspace - please retry or consider unchecking 'include notebooks'".

Another issue both here and in Create is the possible orphaned data being created in Firecloud and GCS. This could become a problem down the road, so we should consider trying to have some asynchronous approach in place for identifying these orphans and clearing them out. One option could be to write a primordial `Workspace` row in AoU before doing anything else, then transitioning it to an `ACTIVE` state once we've successfully initialized things in Firecloud. A query by activation state + creation time would yield the orphans which we could automatically clean up. We could also have a similar mechanism if we wanted to support soft deletion.